test/CoWSwapEthFlow.t.sol

```diff
@@ -25,13 +26,47 @@ contract EthFlowTestSetup is Test {
     }
 
     CoWSwapEthFlowExposed internal ethFlow;
-    IERC20 internal wrappedNativeToken =
-        IERC20(0x1234567890123456789012345678901234567890);
+    IWrappedNativeToken internal wrappedNativeToken =
+        IWrappedNativeToken(0x1234567890123456789012345678901234567890);
     ICoWSwapSettlement internal cowSwap =
         ICoWSwapSettlement(Constants.COWSWAP_ADDRESS);
+    address internal vaultRelayer = 0x0987654321098765432109876543210987654321;
 
     function setUp() public {
+        vm.mockCall(
+            address(cowSwap),
+            abi.encodeWithSelector(ICoWSwapSettlement.vaultRelayer.selector),
+            abi.encode(vaultRelayer)
+        );
+        vm.mockCall(
+            address(wrappedNativeToken),
+            abi.encodeWithSelector(
+                IERC20.approve.selector,
+                vaultRelayer,
+                type(uint256).max
+            ),
+            abi.encode(true)
+        );
         ethFlow = new CoWSwapEthFlowExposed(cowSwap, wrappedNativeToken);
+        vm.clearMockedCalls();
+    }
+
+    function mockAndExpectCall(
+        address to,
+        uint256 value,
+        bytes memory data,
+        bytes memory output
+    ) internal {
+        vm.mockCall(to, value, data, output);
+        vm.expectCall(to, value, data);
+    }
+
+    function mockAndExpectCall(
+        address to,
+        bytes memory data,
+        bytes memory output
+    ) internal {
+        mockAndExpectCall(to, 0, data, output);
     }
```

Comment:
I guess its allowed to have the same function name twice with different input parameters (aka method overrides). Might be difficult for the type hinter to make suggestions.